audio-worklet/src/noisegate-script-processor.js

```diff
@@ -108,11 +108,11 @@ class NoiseGate {
 
     let envelope = this.detectLevel_(this.channel_);
     envelope = this.convertEnvelopeToSineNormalizedDecibelScale_(envelope);
-
+    let weights = this.computeWeights_(envelope);
```

Comment:
It seems my response also got swallowed in the review system!

"I moved the weight computation outside the loop, but I left the scaling in a separate function.
I think its cleaner if computeWeights_ function does not alter its input when setting the weights. I also independently test computeWeights_ in the test code, and modifying the envelope in the computeWeights_ function would obfuscate that code."